### PR TITLE
Feature/lagom unique courseIDs

### PR DESCRIPTION
**Pull request by ldklenner**
### Reason for this PR:
- API change: courseID is now a string

### Changes in this PR:
- Adjust lagom implementation to handle string courseID

### CODE REVIEW

File: product_code/lagom/course_service/impl/src/main/scala/de/upb/cs/uc4/course/impl/CourseServiceImpl.scala [hunk 1]
```diff
@@ -50,22 +50,25 @@ class CourseServiceImpl(clusterSharding: ClusterSharding,
...
-    ref.ask[Confirmation](replyTo => CreateCourse(courseToAdd, replyTo))
-      .map {
-        case Accepted => // Creation Successful
-          (ResponseHeader(201, MessageProtocol.empty, List(("1","Operation successful"))),Done)
-        case Rejected(reason) => // Already exists
-          (ResponseHeader(409,  MessageProtocol.empty, List(("1",reason))),Done)
-      }
+    (_,courseProposal) =>
+      // Generate unique ID for the course to add
+      val courseToAdd = courseProposal.copy(courseId = com.datastax.driver.core.utils.UUIDs.timeBased.toString)
```
Comment (david-buderus) [inline product_code/lagom/course_service/impl/src/main/scala/de/upb/cs/uc4/course/impl/CourseServiceImpl.scala:55]: Why not as import?

File: product_code/lagom/course_service/impl/src/main/scala/de/upb/cs/uc4/course/impl/CourseServiceImpl.scala [hunk 2]
```diff
@@ -28,7 +28,7 @@ class CourseServiceImpl(clusterSharding: ClusterSharding,
   readSide.register(processor)
 
   /** Looks up the entity for the given ID */
-  private def entityRef(id: Long): EntityRef[CourseCommand] =
+  private def entityRef(id: String): EntityRef[CourseCommand] =
     clusterSharding.entityRefFor(CourseState.typeKey, id.toString)
```
Comment (david-buderus) [inline product_code/lagom/course_service/impl/src/main/scala/de/upb/cs/uc4/course/impl/CourseServiceImpl.scala:32]: toString on id is unnecessary

File: product_code/lagom/course_service/impl/src/main/scala/de/upb/cs/uc4/course/impl/CourseServiceImpl.scala (hunk 1 above)
Comment (ldklenner) [reply product_code/lagom/course_service/impl/src/main/scala/de/upb/cs/uc4/course/impl/CourseServiceImpl.scala]: Changed

File: product_code/lagom/course_service/impl/src/main/scala/de/upb/cs/uc4/course/impl/CourseServiceImpl.scala (hunk 2 above)
Comment (ldklenner) [reply product_code/lagom/course_service/impl/src/main/scala/de/upb/cs/uc4/course/impl/CourseServiceImpl.scala]: Changed

**Pull request merged**